akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -383,29 +382,20 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
     byteBuffer.clear()
 
     // Write fixed length parts
-    byteBuffer.put(header.version)
-    byteBuffer.put(header.flags)
-    // 1 empty byte slot // TODO avoid having these empty slots
-    // 1 empty byte slot
-    byteBuffer.position(UidOffset) // skips the above 2 empty slots
-    byteBuffer.putLong(header.uid)
-    byteBuffer.putInt(header.serializer)
+    byteBuffer.put(VersionOffset, header.version)
```

Comment:
yeah, I always wondered why we did it differently for those first fields than the later
is there a performance penalty of using the offset parameter?